node/src/blockchain/test_utils.rs

```diff
@@ -221,31 +241,80 @@ impl Transport for TestTransport {
     }
 }
 
+impl BatchTransport for TestTransport {
+    type Batch = web3::Result<Vec<Result<rpc::Value, web3::Error>>>;
+
+    fn send_batch<T>(&self, requests: T) -> Self::Batch
+    where
+        T: IntoIterator<Item = (RequestId, rpc::Call)>,
+    {
+        self.send_batch_params
+            .lock()
+            .unwrap()
+            .push(requests.into_iter().collect());
+        let response = self.send_batch_results.borrow_mut().remove(0);
+        Box::new(futures::finished(response))
+    }
+}
+
 impl TestTransport {
+    pub fn prepare_params(mut self, params: &Arc<Mutex<Vec<(String, Vec<rpc::Value>)>>>) -> Self {
+        self.prepare_params = params.clone();
+        self
+    }
+
+    pub fn send_result(self, rpc_call_response: rpc::Value) -> Self {
+        self.send_results.borrow_mut().push_back(rpc_call_response);
+        self
+    }
+
+    pub fn send_batch_params(
+        mut self,
+        params: &Arc<Mutex<Vec<Vec<(RequestId, rpc::Call)>>>>,
+    ) -> Self {
+        self.send_batch_params = params.clone();
+        self
+    }
+
+    pub fn send_batch_result(
+        self,
+        batched_responses: Vec<Result<rpc::Value, web3::Error>>,
+    ) -> Self {
+        self.send_batch_results.borrow_mut().push(batched_responses);
+        self
+    }
+
+    pub fn initiate_reference_counter(mut self, reference_arc: &Arc<()>) -> Self {
```

Comment:
Also please delete this method. Your card doesn't need it and I don't want to let it be in master until there is actual usage for it. :slightly_smiling_face:

Reply:
Deleted